src/features/commentsSlice.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable prettier/prettier */
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { createComment, deleteComment, getPostComments } from '../api/comments';
+import { Comment, CommentData } from '../types/Comment';
+import { RootState } from '../app/store';
+
+type CommentsSlice = {
+  comments: Comment[];
+  isLoading: boolean;
+  hasError: boolean;
+  isSubmiting: boolean;
+};
+
+const initialState: CommentsSlice = {
+  comments: [] as Comment[],
```

Comment:
```suggestion
  comments: [],
```
Avoid using `as` it's a trick for typescript that can cause issues